### ISSUE TITLE
mkstemp issues in pytest

### ISSUE THREAD

**Issue by s-mandra** [ngnrsaa/qflex#174]
```
python3 -m pytest tests/python/simulation_test.py 
Test session starts (platform: linux, Python 3.7.5, pytest 5.2.1, pytest-sugar 0.9.2)
rootdir: /home/dagart/NASA_Local/repository/qflex
plugins: sugar-0.9.2
collecting ... 
 tests/python/simulation_test.py ✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓                                         100% ██████████
===================================================================================== warnings summary ======================================================================================
/home/dagart/.local/lib/python3.7/site-packages/cirq/google/engine/engine.py:32
  /home/dagart/.local/lib/python3.7/site-packages/cirq/google/engine/engine.py:32: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Iterable

-- Docs: https://docs.pytest.org/en/latest/warnings.html

Results (6.37s):
     100 passed
Exception ignored in: <function QFlexCircuit.__del__ at 0x7fe311b9f200>
Traceback (most recent call last):
  File "/home/dagart/NASA_Local/repository/qflex/python/cirq_interface/qflex_circuit.py", line 84, in __del__
TypeError: 'NoneType' object is not callable
Exception ignored in: <function QFlexOrder.__del__ at 0x7fe311b98f80>
Traceback (most recent call last):
  File "/home/dagart/NASA_Local/repository/qflex/python/cirq_interface/qflex_order.py", line 52, in __del__
TypeError: 'NoneType' object is not callable
Exception ignored in: <function QFlexGrid.__del__ at 0x7fe311d19e60>
Traceback (most recent call last):
  File "/home/dagart/NASA_Local/repository/qflex/python/cirq_interface/qflex_grid.py", line 60, in __del__
TypeError: 'NoneType' object is not callable
```

**Comment by kkissell** [comment]
This is part of what I believe I fixed in the patch in #289

**Comment by alexandrupaler** [comment]
closed for the moment, because temp files are handled correctly.
